lib/chef/knife/org_show.rb

```diff
@@ -15,19 +15,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-require_relative "../mixin/root_rest"
 
 class Chef
   class Knife
     class OrgShow < Knife
       category "CHEF ORGANIZATION MANAGEMENT"
       banner "knife org show ORGNAME"
 
-      include Chef::Mixin::RootRestv0
+      deps do
+        require_relative "../org"
+      end
 
       def run
         org_name = @name_args[0]
-        ui.output root_rest.get("organizations/#{org_name}")
+        org = Chef::Org.from_hash({ "name" => org_name })
+        ui.output org.chef_rest.get("organizations/#{org_name}")
```

Comment:
This seems like this should be pushed into the Chef::Org object.

I haven't looked but we should be able to do something to get a Chef::Org object for the org_name directly rather than going in through `from_hash` and then the `get` call should be hidden from the caller.   That could be combined into one static method if necessary like `Chef::Org.load(org_name)` (unless that collides with an existing method name but something like that)

Reply:
okay I will look into that

Reply:
@lamont-granquist I would prefer to add `root_rest` method to `knife.rb` similar to `rest` so that it will be available to all subclasses as we have more root level operations into chef core.

```
    def root_rest
      @root_rest ||= begin
        require_relative "server_api"
        Chef::ServerAPI.new(Chef::Config[:chef_server_root])
      end
    end
```

Reply:
So the point here though is that AFAIK we should not be using protocol v0, and I don't understand why this PR is spreading it everywhere.

The chef/user.rb that defines the chef_rest_v0 object has a large banner at the top of it that Chef::User should be replaced by Chef::UserV1 in Chef 13 (that didn't happen) but AFAIK it should all go away.  I have no idea why knife-opc was using v0, that may have been due to some kind of backwards compatibility with now-ancient chef-servers.  Without knowing what breaks, my position is that it should be removed and we should be using our normal v2 rest objects consistently.

Reply:
knife-opc was using v0 simply because we needed it to work ASAP and that was the smallest possible change, so :+1: to doing it right this time around.